cpp/memilio/epidemiology/lct_infection_state.h

```diff
@@ -31,12 +31,12 @@ namespace mio
  * @tparam Ns Number of subcompartments for each infection state defined in InfectionState. 
  *      The number of given template arguments must be equal to the entry Count from InfectionState.
  */
-template <class InfectionStates, unsigned int... Ns>
+template <class InfectionStates, int... Ns>
 class LctInfectionState
 {
 public:
     using InfectionState = InfectionStates;
-    static_assert((unsigned int)InfectionState::Count == sizeof...(Ns),
+    static_assert((int)InfectionState::Count == sizeof...(Ns),
```

Comment:
sizeof returns a size_t, so you should not cast to int here
```suggestion
    static_assert((size_t)InfectionState::Count == sizeof...(Ns),
```